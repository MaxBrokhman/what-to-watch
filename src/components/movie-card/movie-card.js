import React from 'react';

import {Videoplayer} from '../videoplayer/videoplayer';
// eslint-disable-next-line
export const MovieCard = ({movie, hoverHandler, leaveHandler, activeCard}) => (
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={hoverHandler(movie)}
    onMouseLeave={leaveHandler}
  >
    <div className="small-movie-card__image">
      {
        movie === activeCard
          ? <Videoplayer
            // eslint-disable-next-line
            src={movie.preview_video_link}
            width="280"
            height="175"
            autoPlay
            muted
          />
          // eslint-disable-next-line
          : <img src={movie.preview_image} alt={movie.name} width="280" height="175" />
      }
    </div>
    <h3 className="small-movie-card__title">
      {// eslint-disable-next-line
      <a className="small-movie-card__link" href="movie-page.html">{movie.name}</a>}
    </h3>
  </article>
);
