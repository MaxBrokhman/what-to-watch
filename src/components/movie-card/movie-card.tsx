import React from 'react';

import {Videoplayer} from '../videoplayer/videoplayer';
import {Link} from 'react-router-dom';
import {TFilm} from '../../reducer/types';

type TMovieCardProps = {
  movie: TFilm;
  hoverHandler: (movie: TFilm) => () => void;
  leaveHandler: () => void;
  activeCard: TFilm;
}

export const MovieCard = ({
  movie,
  hoverHandler,
  leaveHandler,
  activeCard,
}: TMovieCardProps): JSX.Element => (
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={hoverHandler(movie)}
    onMouseLeave={leaveHandler}
  >
    <Link to={`/movie-card/${movie.id}`}>
      <div className="small-movie-card__image">
        {
          movie === activeCard
            ? <Videoplayer
              src={movie.preview_video_link}
              width="280"
              height="175"
              autoPlay
              muted
            />
            : <img
              src={movie.preview_image}
              alt={movie.name}
              width="280"
              height="175"
            />
        }
      </div>
    </Link>
    <h3 className="small-movie-card__title">
      <a className="small-movie-card__link" href="movie-page.html">{movie.name}</a>
    </h3>
  </article>
);
