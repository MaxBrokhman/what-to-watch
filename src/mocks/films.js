/* eslint-disable */
export const films = [
  {
    name: `Fantastic Beasts: The Crimes of Grindelwald`,
    preview_image: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    preview_video_link: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    genre: `fantasy`,
    background_image: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    poster_image: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    run_time: 88,
    released: 2003,
    id: 1,
  },
  {
    name: `Bohemian Rhapsody`,
    preview_image: `img/bohemian-rhapsody.jpg`,
    preview_video_link: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    genre: `drama`,
    background_image: `img/bohemian-rhapsody.jpg`,
    poster_image: `img/bohemian-rhapsody.jpg`,
    run_time: 88,
    released: 2003,
    id: 2,
  },
  {
    name: `Macbeth`,
    preview_image: `img/macbeth.jpg`,
    preview_video_link: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    genre: `drama`,
    background_image: `img/macbeth.jpg`,
    poster_image: `img/macbeth.jpg`,
    run_time: 88,
    released: 2003,
    id: 3,
  },
  {
    name: `Aviator`,
    preview_image: `img/aviator.jpg`,
    preview_video_link: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    genre: `thriller`,
    background_image: `img/aviator.jpg`,
    poster_image: `img/aviator.jpg`,
    run_time: 88,
    released: 2003,
    id: 4,
  },
  {
    name: `We need to talk about Kevin`,
    preview_image: `img/we-need-to-talk-about-kevin.jpg`,
    preview_video_link: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    genre: `comedy`,
    background_image: `img/we-need-to-talk-about-kevin.jpg`,
    poster_image: `img/we-need-to-talk-about-kevin.jpg`,
    run_time: 88,
    released: 2003,
    id: 5
  }
];
