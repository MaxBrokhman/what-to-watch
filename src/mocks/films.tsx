/* eslint-disable */
import {TFilm} from "../reducer/types";

export const films: Array<TFilm> = [
  {
    name: `Fantastic Beasts: The Crimes of Grindelwald`,
    preview_image: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    preview_video_link: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    genre: `fantasy`,
    background_image: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    poster_image: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    run_time: 88,
    released: 2003,
    id: 1,
    director: ``,
    starring: [],
    rating: 9,
    description: ``,
  },
  {
    name: `Bohemian Rhapsody`,
    preview_image: `img/bohemian-rhapsody.jpg`,
    preview_video_link: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    genre: `drama`,
    background_image: `img/bohemian-rhapsody.jpg`,
    poster_image: `img/bohemian-rhapsody.jpg`,
    run_time: 88,
    released: 2003,
    id: 2,
    director: ``,
    starring: [],
    rating: 9,
    description: ``,
  },
  {
    name: `Macbeth`,
    preview_image: `img/macbeth.jpg`,
    preview_video_link: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    genre: `drama`,
    background_image: `img/macbeth.jpg`,
    poster_image: `img/macbeth.jpg`,
    run_time: 88,
    released: 2003,
    id: 3,
    director: ``,
    starring: [],
    rating: 9,
    description: ``,
  },
  {
    name: `Aviator`,
    preview_image: `img/aviator.jpg`,
    preview_video_link: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    genre: `thriller`,
    background_image: `img/aviator.jpg`,
    poster_image: `img/aviator.jpg`,
    run_time: 88,
    released: 2003,
    id: 4,
    director: ``,
    starring: [],
    rating: 9,
    description: ``,
  },
  {
    name: `We need to talk about Kevin`,
    preview_image: `img/we-need-to-talk-about-kevin.jpg`,
    preview_video_link: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    genre: `comedy`,
    background_image: `img/we-need-to-talk-about-kevin.jpg`,
    poster_image: `img/we-need-to-talk-about-kevin.jpg`,
    run_time: 88,
    released: 2003,
    id: 5,
    director: ``,
    starring: [],
    rating: 9,
    description: ``,
  }
];
