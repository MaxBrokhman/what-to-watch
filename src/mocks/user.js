export const user = {
  id: 1,
  email: `Oliver.conner@gmail.com`,
  name: `Oliver`,
  // eslint-disable-next-line
  avatar_url: `#`,
};
