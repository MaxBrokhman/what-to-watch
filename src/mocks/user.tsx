import {TUser} from "../reducer/types";

export const user: TUser = {
  email: `Oliver.conner@gmail.com`,
  name: `Oliver`,
  // eslint-disable-next-line
  avatar_url: `#`,
};
