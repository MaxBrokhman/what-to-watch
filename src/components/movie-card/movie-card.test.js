import React from 'react';
import renderer from 'react-test-renderer';
import {BrowserRouter} from 'react-router-dom';

import {MovieCard} from './movie-card';
import {films} from '../../mocks/films';

it(`Renders MovieCard component correctly`, () => {
  const hoverHandler = jest.fn();
  const leaveHandler = jest.fn();
  const tree = renderer
                .create(
                    <BrowserRouter>
                      <MovieCard
                        hoverHandler={hoverHandler}
                        movie={films[0]}
                        leaveHandler={leaveHandler}
                        activeCard={null}
                      />
                    </BrowserRouter>
                )
                .toJSON();
  expect(tree).toMatchSnapshot();
});
