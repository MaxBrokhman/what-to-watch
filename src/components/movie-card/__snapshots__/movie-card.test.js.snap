// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders MovieCard component correctly 1`] = `
<article
  className="small-movie-card catalog__movies-card"
  onMouseLeave={[MockFunction]}
>
  <a
    href="/movie-card/1"
    onClick={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="Fantastic Beasts: The Crimes of Grindelwald"
        height="175"
        src="img/fantastic-beasts-the-crimes-of-grindelwald.jpg"
        width="280"
      />
    </div>
  </a>
  <h3
    className="small-movie-card__title"
  >
    <a
      className="small-movie-card__link"
      href="movie-page.html"
    >
      Fantastic Beasts: The Crimes of Grindelwald
    </a>
  </h3>
</article>
`;
